src/app/revenue-share/revenue-share.component.ts

```diff
@@ -1,26 +1,100 @@
-import { displayErrorMessage } from '../utilities/error-handler';
+import { ProjectService } from '../projects/project-service';
+import { SpinnerUtil } from '../utilities/spinner-utilities';
+import { hideSpinnerAndDisplayError } from '../utilities/error-handler';
+import { autonumericCurrency, stripCurrencyData } from '../utilities/currency-util';
+import swal from 'sweetalert2';
+import { ArkaneConnect, SecretType, SignatureRequestType, WindowMode } from '@arkane-network/arkane-connect';
+import { ManagePaymentsService } from '../project/manage-payments/manage-payments.service';
+import { BroadcastService } from '../broadcast/broadcast-service';
 
 @Component({
-  selector: 'app-revenue-share',
-  templateUrl: './revenue-share.component.html',
-  styleUrls: ['./revenue-share.component.css']
+    selector: 'app-revenue-share',
+    templateUrl: './revenue-share.component.html',
+    styleUrls: ['./revenue-share.component.css']
 })
 export class RevenueShareComponent implements OnInit {
+    @Input() projectName: string;
+    @Input() amountInvested: number;
+    @Input() amountInvestedConfirm: number;
+    @ViewChild('modalConfirmPopupDialog') modalPopupDialog: ElementRef;
 
-  projectID: string;
-  amount: number;
+    projectID: string;
 
-  constructor(private route: ActivatedRoute) { }
+    constructor(private route: ActivatedRoute,
+                private projectService: ProjectService,
+                private managePaymentService: ManagePaymentsService,
+                private broadcastService: BroadcastService) {
+    }
 
-  ngOnInit() {
-    this.fetchDataFromRoute()
-  }
+    ngOnInit() {
+        this.fetchDataFromRoute();
+        this.formatConfirmPopupInputField();
+    }
 
-  fetchDataFromRoute() {
-    this.projectID = this.route.snapshot.params.projectID;
-    this.amount = this.route.snapshot.params.amount;
-  }
+    fetchDataFromRoute() {
+        this.projectID = this.route.snapshot.params.projectID;
+        this.amountInvested = this.route.snapshot.params.amount;
+        this.getProject(this.projectID);
+    }
 
+    getProject(projectUUID: string) {
+        SpinnerUtil.showSpinner();
+        this.projectService.getProject(projectUUID)
+            .subscribe(res => {
+                SpinnerUtil.hideSpinner();
+                this.projectName = res.name;
+            }, error => {
+                hideSpinnerAndDisplayError(error);
+            });
+    }
+
+    confirmButtonClicked() {
+        const revenueAmountInvested = parseInt(stripCurrencyData(String(this.amountInvested)), 10);
+        const revenueAmountInvestedConfirm = parseInt(stripCurrencyData(String(this.amountInvestedConfirm)), 10);
+
+        if (revenueAmountInvested !== revenueAmountInvestedConfirm) {
+            swal('', 'The revenue share amounts don\'t match. Please check the proper amount and try again!',
+                'error').then(() => {
+                (<any>$('#modal-confirm-revenue')).modal('hide');
```

Comment:
By using modal component, you won't need this jQuery manipulation inside Angular code.